config/7.0/obdemo-bank/ig/scripts/groovy/ProcessRegistration.groovy

```diff
@@ -96,44 +109,54 @@ def scopes = oidcRegistration.getClaim("scope")
 def roles = attributes.clientCertificate.roles
 
 if (scopes.contains(SCOPE_ACCOUNTS) && !(roles.contains(ROLE_ACCOUNT_INFORMATION))) {
-    logger.error("Requested scope {} requires certificate role {}",
-            SCOPE_ACCOUNTS,
-            ROLE_ACCOUNT_INFORMATION
-    )
-    return new Response(Status.FORBIDDEN)
+    message = "Requested scope " + SCOPE_ACCOUNTS + " requires certificate role " + ROLE_ACCOUNT_INFORMATION
+    logger.error(message)
+    response.status = Status.FORBIDDEN
+    response.entity = "{ \"error\":\"" + message + "\"}"
+    return response
 }
 
 if (scopes.contains(SCOPE_PAYMENTS) && !(roles.contains(ROLE_PAYMENT_INITIATION))) {
-    logger.error("Requested scope {} requires certificate role {}",
-            SCOPE_PAYMENTS,
-            ROLE_PAYMENT_INITIATION
-    )
-    return new Response(Status.FORBIDDEN)
+    message = "Requested scope " + SCOPE_PAYMENTS + " requires certificate role " + ROLE_PAYMENT_INITIATION
+    logger.error(message)
+    response.status = Status.FORBIDDEN
+    response.entity = "{ \"error\":\"" + message + "\"}"
+    return response
 }
 
 // Cross check ID with cert
 //
-// e.g. PSDGB-FFA-5f563e89742b2800145c7da1
+// e.g. PSDGB-FFA-5f563e89742b2800145c7da1 or PSDGB-OB-Unknown0015800001041REAAY (issue by OB)
 
 def  organizationalIdentifier = attributes.clientCertificate.subjectDNComponents.OI
 
 if (!organizationalIdentifier) {
-    logger.error("No organizational identifier in cert")
-    return new Response(Status.FORBIDDEN)
+    message = "No organizational identifier in cert"
+    logger.error(message)
+    response.status = Status.FORBIDDEN
+    response.entity = "{ \"error\":\"" + message + "\"}"
+    return response
 }
 
 def oiComponents = organizationalIdentifier.split("-")
 
-if (oiComponents.length != 3) {
-    logger.error("Wrong number of dashes in OI {} - expected 2",organizationalIdentifier)
-    return new Response(Status.FORBIDDEN)
+if (oiComponents.length > 3) {
+    message = "Wrong number of dashes in OI " + organizationalIdentifier +" - expected 2"
+    logger.error(message)
+    response.status = Status.FORBIDDEN
+    response.entity = "{ \"error\":\"" + message + "\"}"
+    return response
 }
 
-def dnId = oiComponents[2]
+// Issue: https://github.com/SecureBankingAccessToolkit/securebanking-openbanking-demo/issues/53
+def dnId = oiComponents[2].toString().replace("Unknown","")
```

Comment:
The trick to fix the issue related with the 'Unknown' value in the organisation identifier, this will work for all values cases.